std/variant.d

```diff
@@ -700,7 +700,15 @@ public:
                 {
                     alias UT = Unqual!T;
                     auto p = new UT;
-                    *p = rhs;
+                    static if (isAssignable!UT)
+                    {
+                        *p = rhs;
+                    }
+                    else
+                    {
+                        import core.lifetime : emplace;
+                        emplace(p, rhs);
```

Comment:
AFAIK, this requires the type `UT` to feature a constructor taking a `T rhs` lvalue argument, and should thus be equivalent to `auto p = new UT(rhs)`. - I think that's not suited in general and should probably be replaced by a manual copy-emplace as used for the `T.sizeof <= size` case above. It cheats around immutability by faking a copy-construct, performing a bitcopy and invoking the postblit if necessary. By the looks of it, it currently seems to ignore copy ctors...

Reply:
=> https://github.com/dlang/druntime/pull/3239